wgpu-types/src/lib.rs

```diff
@@ -2109,6 +2037,467 @@ pub enum TextureFormat {
+                    "etc2-rgba8unorm-srgb" => TextureFormat::Etc2Rgba8UnormSrgb,
+                    "eac-r11unorm" => TextureFormat::EacR11Unorm,
+                    "eac-r11snorm" => TextureFormat::EacR11Snorm,
+                    "eac-rg11unorm" => TextureFormat::EacRg11Unorm,
+                    "eac-rg11snorm" => TextureFormat::EacRg11Snorm,
+                    other => {
+                        if other.starts_with("astc") {
+                            let mut parts = other.split('-');
+                            parts.next(); // discard "astc"
+                            let block = parts
+                                .next()
+                                .ok_or_else(|| E::invalid_value(Unexpected::Str(s), &self))?;
+                            let channel = parts.collect::<Vec<&str>>().join("-");
+
+                            let block = match block {
+                                "4x4" => AstcBlock::B4x4,
+                                "5x4" => AstcBlock::B5x4,
+                                "5x5" => AstcBlock::B5x5,
+                                "6x5" => AstcBlock::B6x5,
+                                "6x6" => AstcBlock::B6x6,
+                                "8x5" => AstcBlock::B8x5,
+                                "8x6" => AstcBlock::B8x6,
+                                "8x8" => AstcBlock::B8x8,
+                                "10x5" => AstcBlock::B10x5,
+                                "10x6" => AstcBlock::B10x6,
+                                "10x8" => AstcBlock::B10x8,
+                                "10x10" => AstcBlock::B10x10,
+                                "12x10" => AstcBlock::B12x10,
+                                "12x12" => AstcBlock::B12x12,
+                                _ => return Err(E::invalid_value(Unexpected::Str(s), &self)),
+                            };
+
+                            let channel = match channel.as_ref() {
+                                "unorm" => AstcChannel::Unorm,
+                                "unorm-srgb" => AstcChannel::UnormSrgb,
+                                "hdr" => AstcChannel::Hdr,
+                                _ => return Err(E::invalid_value(Unexpected::Str(s), &self)),
+                            };
+
+                            TextureFormat::Astc { block, channel }
+                        } else {
+                            return Err(E::invalid_value(Unexpected::Str(s), &self));
+                        }
+                    }
+                };
+
+                Ok(format)
+            }
+        }
+
+        deserializer.deserialize_any(TextureFormatVisitor)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl Serialize for TextureFormat {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match *self {
+            TextureFormat::R8Unorm => {
+                let name = "r8unorm";
+                serializer.serialize_unit_variant(name, 0, name)
```

Comment:
This seems wrong to me. The `Deserialize` impl expects a string, so the `Serialize` impl should give one. If not, `deserialize(serialize(format))` won't give the same format back but an error.

Morover, serializing everything as a string better matches the spec, is clearer and more resilient to changes to this enum.

Reply:
@a1phyr this has been addressed now, can you please take a look again?